src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -191,6 +193,10 @@ public function setParameter($name, $value)
      */
     public function set($id, $service, $scope = self::SCOPE_CONTAINER)
     {
+        if (!in_array($scope, array('container', 'request')) || ('request' === $scope && 'request' !== $id)) {
```

Comment:
I'm not sure about this condition, maybe something like `'request' !== $id` is enough.